src/retrace-server-bugzilla-query

```diff
@@ -47,37 +48,42 @@ if __name__ == "__main__":
         else:
             log.write("Not logged in. Continue as anonymous user.\n")
 
+        found_tasks: Dict[str, List[str]] = {}
         product_list = CONFIG.get_list("BugzillaProduct")
         component_list = CONFIG.get_list("BugzillaComponent")
-
-        query = bzapi.build_query(
-            product=product_list,
-            component=component_list,
-            include_fields=["id"])
-
-        # I don't want to run regex on each and every comment, at first try fast check
-        # if trigger word is in comment and if is, then use slower regex
+        config_status = CONFIG.get_list("BugzillaStatus")
         trigger_words = CONFIG.get_list("BugzillaTriggerWords")
         regexes = CONFIG.get_list("BugzillaRegExes")
+        if CONFIG["BugzillaQueryLastChangeDelta"]:
+            delta = CONFIG["BugzillaQueryLastChangeDelta"]
+        else:
+            delta = 7*24
 
-        config_status = CONFIG.get_list("BugzillaStatus")
+        query = bzapi.build_query(product=product_list,
+                                  component=component_list,
+                                  include_fields=["id", "comments"])
         query["status"] = list(set(BUGZILLA_STATUS).difference(config_status))
-
-        found_tasks: Dict[str, List[str]] = {}
-
-        bugs = [x.bug_id for x in bzapi.query(query)]
-
-        while bugs:
-            query = bzapi.build_query(
-                bug_id=bugs[:250],
-                include_fields=["id", "comments"])
-
-            cur_bugs = bzapi.query(query)
-            bugs = bugs[250:]
-            for bug in cur_bugs:
+        query["last_change_time"] = datetime.today() - timedelta(hours = delta)
+        bugzilla_offset = 0
+        bugzilla_bug_count = 0
+        while True:
+            bugzilla_offset += bugzilla_bug_count
+            query["offset"] = bugzilla_offset
+            query["limit"] = 500
```

Comment:
I agree we don't need a config variable for the limit, but could you please pull the 500 out into a named constant at the top of the file? Just so we don't introduced any new magical numbers.

Reply:
Yes